### PR TITLE
Shellcheck compliant

### PR DESCRIPTION
**Pull request by fvaleri**
Make log-dump and cold-backup scripts compliant with shellcheck, and add them to the shellcheck script in our CI.

### CODE REVIEW

File: .azure/scripts/shellcheck.sh [hunk 1]
```diff
@@ -12,7 +12,9 @@ docker-images/kafka-based/kafka/stunnel-scripts/*.sh
 docker-images/kafka-based/kafka/exporter-scripts/*.sh
 docker-images/jmxtrans/*.sh
 tools/olm-bundle/*.sh
-tools/*.sh"
+tools/*.sh
+tools/cold-backup/run.sh
+tools/log-dump/run.sh"
```
Comment (scholzj) [inline .azure/scripts/shellcheck.sh:17]: Should this instead of `run.sh` be `*.sh` to cover any scripts added int he future?

File: systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java [hunk 2]
```diff
@@ -36,11 +35,13 @@
 @Tag(INTERNAL_CLIENTS_USED)
 @IsolatedSuite
 public class ColdBackupScriptIsolatedST extends AbstractST {
-
     private static final Logger LOGGER = LogManager.getLogger(ColdBackupScriptIsolatedST.class);
 
     @IsolatedTest
     void backupAndRestore(ExtensionContext context) {
+        clusterOperator.unInstall();
+        clusterOperator = clusterOperator.defaultInstallation().createInstallation().runInstallation();
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: Why was this change needed?

CC @see-quick

File: tools/cold-backup/README.md [hunk 3]
```diff
@@ -31,25 +29,22 @@ kubectl create -f ./examples/kafka/kafka-persistent.yaml
...
+  bin/kafka-consumer-groups.sh --bootstrap-server :9092 --group my-group --describe > /tmp/offsets.txt
 
 # send additional 12345 messages
-kubectl run kafka-producer-perf-test -it --image="$CLIENT_IMAGE" \
-  --rm="true" --restart="Never" -- bin/kafka-producer-perf-test.sh \
-  --topic my-topic --record-size 1000 --num-records 12345 --throughput -1 \
-  --producer-props acks=1 bootstrap.servers="my-cluster-kafka-bootstrap:9092"
+kubectl exec -it my-cluster-kafka-0 -c kafka -- \
+  bin/kafka-producer-perf-test.sh --topic my-topic --record-size 1000 --num-records 12345 \
+  --throughput -1 --producer-props acks=1 bootstrap.servers="my-cluster-kafka-bootstrap:9092"
```
Comment (scholzj) [inline tools/cold-backup/README.md:47]: Why change to `kubectl exec`? That is a bad pattern which should not be encouraged by our docs. It can eaily cause the brokers to get killed etc.

File: tools/cold-backup/run.sh [hunk 4]
```diff
@@ -331,32 +327,37 @@ restore() {
...
   if [[ $CONFIRM == true ]]; then
     echo "Restore of cluster $NAMESPACE/$CLUSTER_NAME"
     confirm
   fi
-  uncompress $SOURCE_FILE $tmp
-  source $tmp/env
+  
+  local tmp="$BACKUP_PATH/$NAMESPACE/$CLUSTER_NAME"
+  uncompress "$SOURCE_FILE" "$tmp"
+  # shellcheck disable=SC1091
```
Comment (scholzj) [inline tools/cold-backup/run.sh:343]: Maybe you can add a short comment why is it needed into the code (I assume it cannot follow the code since it does nto exist)

**Review by scholzj — COMMENTED** [review]
Some nits. LGTM otherwise. I will trigger the regression.

File: systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java (hunk 2 above)
Comment (fvaleri) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: The error I had consistently on ColdBackup ST is "ZK PVC is already created", or something like that. So I just did what I found in the LogDump ST and it worked.

File: tools/cold-backup/README.md (hunk 3 above)
Comment (fvaleri) [reply tools/cold-backup/README.md:47]: Yeah, this is just to avoid updating the image version at every release. I can add a note if you think it is needed.

File: tools/cold-backup/run.sh (hunk 4 above)
Comment (fvaleri) [reply tools/cold-backup/run.sh:343]: https://github.com/koalaman/shellcheck/wiki/SC1091

File: .azure/scripts/shellcheck.sh (hunk 1 above)
Comment (fvaleri) [reply .azure/scripts/shellcheck.sh:17]: OK, but what about any sub dir containing scripts?

File: tools/cold-backup/README.md (hunk 3 above)
Comment (scholzj) [reply tools/cold-backup/README.md:47]: Do we need to update the image every release? I assume it will keep working in general. You can also use a placeholder for the image and let users to fill it in themself.

File: tools/cold-backup/run.sh (hunk 4 above)
Comment (scholzj) [reply tools/cold-backup/run.sh:343]: Well, that is what I assumed. But would be still good to have it in the code.

File: .azure/scripts/shellcheck.sh (hunk 1 above)
Comment (scholzj) [reply .azure/scripts/shellcheck.sh:17]: Well, I didnt said it would be perfect ... but it would be better. I'm not sure if it accepts something like `tools/log-dump/**/*.sh` or soemthing like thate.

File: systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java (hunk 2 above)
Comment (fvaleri) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: I think there has been some changes recently. Looking at the new test name, I assume this runs in isolation. Maybe @see-quick can shed a light on this.

File: tools/cold-backup/README.md (hunk 3 above)
Comment (fvaleri) [reply tools/cold-backup/README.md]: ok

File: tools/cold-backup/run.sh (hunk 4 above)
Comment (fvaleri) [reply tools/cold-backup/run.sh]: ok

File: systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: Additionally, if only one test case will be inside `LogDumpScriptIsolatedST` that's okay and we do not need to do such a thing but practically speaking it would not an ideal...

Comment (fvaleri) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: @see-quick this is the error when I use before all. It happens on restore step. 

```
Error from server (AlreadyExists): error when creating "STDIN": persistentvolumeclaims "data-my-cluster-1435185444-zookeeper-0" already exists
error: no matching resources found
```

[strimzi-debug.log](https://github.com/strimzi/strimzi-kafka-operator/files/8117439/strimzi-debug.log)

Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:43]: Hmmm, I see...in `@BeforeAll` seems to be missing assigning the default installation to the cluster operator instance so instead of this:
```java
@BeforeAll
    void setUp() {
        clusterOperator.unInstall();
        clusterOperator.defaultInstallation().createInstallation().runInstallation();
    }
```
you should correct it to:
```java
@BeforeAll
    void setUp() {
        clusterOperator.unInstall();
        clusterOperator = clusterOperator.defaultInstallation().createInstallation().runInstallation();
    }

Comment (fvaleri) [reply systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:44]: ok, fix pushed

**Review by see-quick — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.
